### PR TITLE
Upd batch

### PR DESCRIPTION
**Pull request by k-dominik**
Batch processing docs were quite outdated, referencing steps that were just not there anymore.

Summary:
* updated images
* adapted text to current (1.4.0) gui state
* added an FAQ, inspired by forum posts we get

fixes https://github.com/ilastik/ilastik.github.io/issues/64

### CODE REVIEW

File: documentation/basics/batch.md [hunk 1]
```diff
@@ -7,27 +7,59 @@ weight: 6
 ---
 # Batch Processing
 
-In the machine learning based ilastik workflows such as the
-[pixel classification workflow]({{site.baseurl}}/documentation/objects/objects.html) and the [density counting workflow]({{site.baseurl}}/documentation/counting/counting.html), the user interactively trains a classifier on a representative set of images. After that training step, the generated classifier can be used for batch processing.
-
-<!-- The following documentation applies to the [Density Counting workflow]({{site.baseurl}}/documentation/counting/counting.html) and to the [Pixel Classification workflow]({{site.baseurl}}/documentation/objects/objects.html). -->
+In the machine learning based ilastik workflows such as the [Pixel Classification Workflow][pixelclass], the user interactively trains a classifier on a representative set of images.
+After that training step, the generated classifier can be applied to more data by using batch processing.
```
Comment (btbest) [inline documentation/basics/batch.md:11]:
```suggestion
In most ilastik workflows, such as the [Pixel Classification Workflow][pixelclass], the user interactively trains a classifier on a representative set of images.
The trained classifier can be applied to more data by using batch processing.
```

File: documentation/basics/batch.md [hunk 2]
```diff
@@ -7,27 +7,59 @@ weight: 6
...
+
+The next step is the actual batch processing itself.
+
+<a href="screenshots/batch_processing_02.png" data-toggle="lightbox"><img src="screenshots/batch_processing_02.png" class="img-responsive" /></a>
+
+After clicking on the **Process all files** button, ilastik begins batch processing all images, and writes the resulting classification result to the specified output files.
+With default export settings, the output files are stored as hdf5 files in the same directory where the input file is located.
+Their file names will be the same as the original files, plus one or two words indicating the export source (e.g. "Probabilities") and the file extension `.h5`.
+
+The exported `.h5` files, contain the resulting prediction as a multidimensional dataset inside the file.
```
Comment (btbest) [inline documentation/basics/batch.md:43]:
```suggestion
The exported `.h5` files contain the resulting prediction as a multidimensional dataset inside the file.
```

File: documentation/basics/batch.md [hunk 3]
```diff
@@ -7,27 +7,59 @@ weight: 6
...
+Further details on this versatile file format (that is easily accessible from Matlab and Python) can be found [here](http://docs.h5py.org/).
```
Comment (btbest) [inline documentation/basics/batch.md:44]:
```suggestion
Further details on this versatile file format (that is easily accessible from Matlab and Python) can be found [in the hdf5 documentation](http://docs.h5py.org/).
```

File: documentation/basics/batch.md [hunk 4]
```diff
@@ -7,27 +7,59 @@ weight: 6
...
 
-The first step in batch prediction is the selection of the input files. This uses the same mechanism as [data selection]({{baseurl}}/documentation/basics/dataselection.html).
+### Batch Processing FAQ
 
-<a href="screenshots/batch1_zoomed.png" data-toggle="lightbox"><img src="screenshots/batch1_zoomed.png" class="img-responsive" /></a>
+#### I have multiple image stacks I want to process, is it possible to load those in batch?
+Image stacks are currently not supported in batch: Whenever your image is spread across multiple files, use the [headless mode][headless], or the [Fiji plugin][fijiplugin] to process those.
```
Comment (btbest) [inline documentation/basics/batch.md:49]:
```suggestion
Stacks with one image file per slice are currently not supported in batch. If your image is spread across multiple files, you can use the [headless mode][headless], or the [Fiji plugin][fijiplugin] to process those.
```

File: documentation/basics/batch.md [hunk 5]
```diff
@@ -7,27 +7,59 @@ weight: 6
...
 
-The next step is the actual batch processing itself. In the **Batch Prediction Output Locations** applet, the user can configure the output format using the **Choose Settings** button. This allow for example to save the results of the workflow as `.png` images rather than `.tiff` or `.h5` files. For novice users the default settings should be fine, however the results of some workflows such as the [density counting workflow]({{site.baseurl}}/documentation/counting/counting.html) cannot be exported in some of these formats (**hdf5 file format is supported by all workflows**).
+#### How do I specify the internal dataset in a `.hdf5`/`.z5`-file?
+It is currently not possible to specify the internal dataset with the graphical user interface, use the [headless mode][headless], or [Fiji plugin](https://github.com/ilastik/ilastik4ij#ilastik-imagej-modules) to process that data.
+Alternatively you could create `hdf5` files with only single virtual dataset each, that point to the file that has multiple ones.
```
Comment (btbest) [inline documentation/basics/batch.md:55]: I don't follow how this suggestion achieves "specifying the internal dataset". Are we asking the user to wrap the batch processing output in another h5?

File: documentation/basics/batch.md [hunk 6]
```diff
@@ -7,27 +7,59 @@ weight: 6
...
 
-<a href="screenshots/batch2_zoomed.png" data-toggle="lightbox"><img src="screenshots/batch2_zoomed.png" class="img-responsive" /></a>
+#### Batch processing is slow, how to make it faster?
+In batch processing, ilastik processes only one file at a time.
+If you have the resources in your machine (number of cores >> 8), you could run multiple instances of ilastik in [headless mode][headless] which work on parts of the full set of images.
```
Comment (btbest) [inline documentation/basics/batch.md:59]:
```suggestion
If you have the resources in your machine (number of cores >> 8), you could run multiple instances of ilastik in [headless mode][headless] and have each instance process a subset of the images.
```

**Review by btbest — APPROVED** [review]
Just some minor phrasing suggestions.

File: documentation/basics/batch.md (hunk 5 above)
Comment (k-dominik) [reply documentation/basics/batch.md:55]: yeah, maybe that's a tad too advanced... So in h5 there is the concept of [virtual datasets](). So what I'm suggesting is to create one for each of the internal paths - such h5 files would appear as if there were only a single dataset. I'll scratch that remark, and wait until such a question actually comes up (in the forum) or we devote some time on improving the ui :)

**Comment by k-dominik** [comment]
thank you @btbest for the great suggestions 🙌

**Pull request merged**